### PR TITLE
vm: interp test + bugfix segfault

### PR DESCRIPTION
**Pull request by 0x0ece**
This includes https://github.com/firedancer-io/firedancer/pull/3528

### CODE REVIEW

File: src/flamenco/runtime/tests/fd_vm_test.c [hunk 1]
```diff
@@ -254,15 +254,15 @@ do{
   vm->reg[8]  = input->vm_ctx.r8;
   vm->reg[9]  = input->vm_ctx.r9;
   // vm->reg[10]  = input->vm_ctx.r10; // set in fd_vm_init
-  vm->reg[11] = input->vm_ctx.r11;
+  // vm->reg[11]  = input->vm_ctx.r11;  // set in fd_vm_init
```
Comment (ravyu-jump) [inline src/flamenco/runtime/tests/fd_vm_test.c:257]: Q: this change is related to the dynamic stack frames, right? Should this be feature/sbpf version gated?

Comment (ravyu-jump) [reply src/flamenco/runtime/tests/fd_vm_test.c]: ah, I see 👍

File: src/flamenco/runtime/tests/fd_vm_test.c [hunk 2]
```diff
@@ -254,15 +254,15 @@ do{
...
+  // vm->reg[11]  = input->vm_ctx.r11; // set in fd_vm_init
 
   // Propagate the acc_regions_meta to the vm
   vm->acc_region_metas = fd_valloc_malloc( valloc, alignof(fd_vm_acc_region_meta_t), sizeof(fd_vm_acc_region_meta_t) * input->vm_ctx.input_data_regions_count );
   setup_vm_acc_region_metas( vm->acc_region_metas, vm, vm->instr_ctx );
 
   // Validate the vm
   if ( fd_vm_validate( vm ) != FD_VM_SUCCESS ) {
-    effects->error = -1;
+    effects->error = -2;
```
Comment (ravyu-jump) [inline src/flamenco/runtime/tests/fd_vm_test.c:265]: nit: comment explaining why might be useful here

File: config/base.mk [hunk 3]
```diff
@@ -5,7 +5,7 @@ SHELL:=bash
 CPPFLAGS:=-isystem ./$(OPT)/include -DFD_LOG_UNCLEAN_EXIT=1 -DFD_HAS_BACKTRACE=0
 RUSTFLAGS:=-C force-frame-pointers=yes
 CC:=gcc
-CFLAGS=-std=c17
+CFLAGS=-std=c17 -fwrapv
```
[the remark this answers is not in the document]
Comment (ravyu-jump) [reply config/base.mk:8]: update: ok looks like this only affects _signed_ overflows, so the ELF loader should be safe at the very least. Not sure about the rest of the codebase

**Review by topointon-jump — DISMISSED** [review]

**Review by topointon-jump — APPROVED** [review]

**Pull request merged**